.github/workflows/ci.yml

```
@@ -74,10 +74,63 @@ jobs:
           DATABASE_URL: postgres://${{ env.PGUSER }}:${{ env.PGUSER }}@${{ env.PGHOST }}:${{ job.services.postgres.ports[5432] }}/${{ env.PGDATABASE }}?sslmode=require
           AWS_SKIP_CREDENTIALS: 1
           AWS_REGION: eu-central-1
+  musl-build:
+    name: Build ${{ matrix.target }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - target: aarch64-unknown-linux-musl
+          - target: x86_64-unknown-linux-musl
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          set-safe-directory: false
+      - uses: taiki-e/install-action@v2
+        with: { tool: cross }
+      - uses: Swatinem/rust-cache@v2
+        if: github.event_name != 'release' && github.event_name != 'workflow_dispatch'
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+        # https://github.com/docker/setup-qemu-action
+        with:
+          platforms: linux/amd64,linux/arm64
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        # https://github.com/docker/setup-buildx-action
+        with:
+          install: true
+          platforms: linux/amd64,linux/arm64
+      - name: Set up AWS SAM
+        uses: aws-actions/setup-sam@v2
+        with:
+          use-installer: true
 
+      - name: Build ${{ matrix.target }}
+        run: |
+          # See https://github.com/cross-rs/cross/issues/1526
+          # TODO: Remove this once a version after cross 0.2.5 is released
+          export CROSS_BUILD_OPTS="--output=type=docker"
+          export "CARGO_TARGET_$(echo ${{ matrix.target }} | tr 'a-z-' 'A-Z_')_RUSTFLAGS"='-C strip=debuginfo'
+          cross build --release --target ${{ matrix.target }} --workspace
+
+      - name: Move the artefacts to the expected place
+        run: |
+          mkdir -p target_releases/${{ matrix.target }}
+          mv target/${{ matrix.target }}/release/martin target_releases/${{ matrix.target }}
+          mv target/${{ matrix.target }}/release/martin-cp target_releases/${{ matrix.target }}
+          mv target/${{ matrix.target }}/release/mbtiles target_releases/${{ matrix.target }}
+      - name: Save build artifacts to build-${{ matrix.target }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-${{ matrix.target }}
+          path: target_releases/*
   docker-build-test:
     name: Build and test docker images
     runs-on: ubuntu-latest
+    needs: [ musl-build, lint-debug-test ]
```

unsure if this should depend on the following instead.
This is kind of the same as `package`, but could also be tests, so this dependency set.

likely, the current version makes CI less dependeent and thus faster, but 🤷‍♂ 


```suggestion
    needs: [ lint-debug-test, test-multi-os, test-with-svc, test-aws-lambda ]
```